pkg/volume/cephfs/cephfs.go

```diff
@@ -299,6 +330,90 @@ func (cephfsVolume *cephfs) execMount(mountpoint string) error {
 	return nil
 }
 
+func (cephfsMounter *cephfsMounter) checkFuseMount() bool {
+	execute := cephfsMounter.plugin.host.GetExec(cephfsMounter.plugin.GetPluginName())
+	switch runtime.GOOS {
+	case "linux":
+		retBytes, err := execute.Run("/bin/ls", "/sbin/mount.fuse.ceph")
+		if err == nil && string(retBytes) == "/sbin/mount.fuse.ceph\n" {
+			glog.V(4).Infof("/sbin/mount.fuse.ceph exists, it should be fuse mount")
+			return true
+		}
+		return false
+	}
+	return false
+}
+
+func (cephfsVolume *cephfs) execFuseMount(mountpoint string) error {
+	// cephfs keyring file
+	keyring_file := ""
+	// override secretfile if secret is provided
+	if cephfsVolume.secret != "" {
+		glog.V(4).Infof("cephfs mount begin using fuse.")
+
+		keyringPath := cephfsVolume.GetKeyringPath()
+		os.MkdirAll(keyringPath, 0750)
+
+		payload := make(map[string]util.FileProjection, 1)
+		var fileProjection util.FileProjection
+
+		keyring := fmt.Sprintf("[client.%s]\n", cephfsVolume.id) + "key = " + cephfsVolume.secret + "\n"
+
+		fileProjection.Data = []byte(keyring)
+		fileProjection.Mode = int32(0644)
+		fileName := cephfsVolume.id + ".keyring"
+
+		payload[fileName] = fileProjection
+
+		writerContext := fmt.Sprintf("cephfuse:%v.keyring", cephfsVolume.id)
+		writer, err := util.NewAtomicWriter(keyringPath, writerContext)
+		if err != nil {
+			glog.Errorf("failed to create atomic writer: %v", err)
+			return err
```

Comment:
cleanup

Reply:
many thanks for review. the cleanup will be in https://github.com/kubernetes/kubernetes/pull/55866/files#diff-021042d70c9aac8994638a8d0827b1e0R242